### PR TITLE
Made changes to catalogue styling and added clickoutsidelistener

### LINKED ISSUE #715: Copy to clipboard tooltip is not aligned / scroll always visible
**Description**

1. `Copy to Clipboard` tooltip is not align properly
2. Scroll always visible.

**Expected Behavior**

1. `Copy to Clipboard` tooltip should be aligned properly.
2. Keep the scroll as `auto` like if the content exceeds then only it will be visible else remains hidden in the modal.

**Screenshots**

https://user-images.githubusercontent.com/44670961/171209794-8cd25020-c7d6-4914-be83-5a4160933c03.mp4

**Enviroment:**
 - OS:  Ubuntu
 - Browser: Chrome, firefox

---
[Optional] **To Reproduce**
1. Go to [Meshery Catalog](https://meshery.io/catalog)
2. Click on any card which will open a modal
3. Hover over `Copy Url` button.

### Contributor Resources

The meshery.io website uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/meshery/meshery.io/tree/master).
- See the [Contributing to Meshery.io Website](https://github.com/meshery/meshery.io#contributing-to-the-mesheryio-website) section of the readme.md and other [contributing instructions](https://docs.meshery.io/project/contributing), too.
- See Meshery site designs in this [Figma designfile](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI?node-id=110%3A1). Fill-in the [Layer5 Community Member Form](https://layer5.io/newcomer) and join the [Layer5 Community Slack](http://slack.layer5.io) for access.

### PR DESCRIPTION
**Pull request by Rohith-Raju**
Signed-off-by: Rohith Raju <rohithraju488@gmail.com>

**Description**
Added flexbox to the catalogue, made it responsive and added clickoutsidelistener feature.

This PR fixes #715 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by asubedy — CHANGES_REQUESTED** [review]
@Rohith-Raju can you add flex box styles to the individual pages too.
image

File: _includes/modal.html [hunk 1]
```diff
@@ -1,35 +1,43 @@
...
-            {%endif%}
-            <h6 class="pattern-type"> {{pattern.type}}</h6>
-            </a>
-        </div>  
-<div class="modal-image" style=" padding-bottom: 19rem;">
-    <img src="{{pattern.image}}" /> 
-    <h4 class="related-patterns">Related Patterns</h4>
-    {% include related-patterns.html %}
-</div>
+  <div class="someclass">
```
Comment (asubedy) [inline _includes/modal.html:2]: Some better name for the class would be good

**Review by asubedy — CHANGES_REQUESTED** [review]

**Comment by asubedy** [comment]
Also is the clickoutsidelistener feature suppose to close the modal when clicked outside the modal? If so that feature is not working maybe some changes to the javascript.

**Comment by Rohith-Raju** [comment]
> Also is the clickoutsidelistener feature suppose to close the modal when clicked outside the modal? If so that feature is not working maybe some changes to the javascript.

@asubedy In the deploy preview, it seems to be working on my end 
https://user-images.githubusercontent.com/55309324/178294083-1ea83508-8c4e-44e9-afd9-394118476b1b.mp4

**Comment by asubedy** [comment]
> > > Also is the clickoutsidelistener feature suppose to close the modal when clicked outside the modal? If so that feature is not working maybe some changes to the javascript.
> > 
> > 
> > @asubedy In the deploy preview, it seems to be working on my end https://user-images.githubusercontent.com/55309324/178294083-1ea83508-8c4e-44e9-afd9-394118476b1b.mp4
> 
> image
> 
> @Rohith-Raju ^ This is the error on the console when clicked outside the modal

Ok, I guess this was because JavaScript was turned off on my browser works fine on other browser, no need to change anything I suppose. 👍

**Review by leecalcote — CHANGES_REQUESTED** [review]
Click away listener still doesn't work on Safari.

**Comment by leecalcote** [comment]
After the inconsequential whitespace/tab formatting is removed, this PR can be accepted, so long as new issues are opened to follow up on outstanding issues.

**Comment by Rohith-Raju** [comment]
@leecalcote I'll make all the changes as per review requests. As I don't have access to safari can it be its own issue

**Comment by leecalcote** [comment]
> @leecalcote I'll make all the changes as per review requests. As I don't have access to safari can it be its own issue

Sound real good.

**Review by asubedy — CHANGES_REQUESTED** [review]
image
@Rohith-Raju please add some space between the left section of the modal and the right section of the modal
decrease the size of the image in the modal slightly and place the image a bit down from the current location.

These changes are not only for wasm modal but also for the pattern modal

**Comment by Rohith-Raju** [comment]
> image @Rohith-Raju please add some space between the left section of the modal and the right section of the modal decrease the size of the image in the modal slightly and place the image a bit down from the current location.
> 
> These changes are not only for wasm modal but also for the pattern modal

@asubedy thanks for that, Some styles were overridden. Gonna have to make separate classes 😄

File: _sass/blog.scss [hunk 2]
```diff
@@ -1,125 +1,124 @@
 .blog-posts {
-	list-style: none;
-	padding: 0;
+  list-style: none;
+  padding: 0;
```
[the remark this answers is not in the document]
Comment (Rohith-Raju) [reply _sass/blog.scss:3]: @leecalcote @asubedy  will it be fine if I just delete the file from the pr?

[the remark this answers is not in the document]
Comment (Rohith-Raju) [reply _sass/blog.scss:3]: @leecalcote @asubedy added the width property and my linter took care of the rest. 😅 
image

File: _sass/catalog.scss [hunk 3]
```diff
@@ -426,11 +683,11 @@
     padding: 20px;
     border-radius: 15px;
     border: 0;
-    background: rgba(0,0,0,0.7) ;
-    box-shadow: 2px 2px 6px 0px  #00D3A9;
+    background: rgba(0, 0, 0, 0.7);
+    box-shadow: 2px 2px 6px 0px #00d3a9;
```
[the remark this answers is not in the document]
Comment (Rohith-Raju) [reply _sass/catalog.scss:687]: Done ✔️ 
image

File: _sass/blog.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (leecalcote) [reply _sass/blog.scss:3]: @Nikhil-Ladha might have a suggestion on how to ensure that lint rules help keep formatting consistent.

[the remark this answers is not in the document]
Comment (Nikhil-Ladha) [reply _sass/blog.scss:3]: Pre-commit-hook is the way to go for this, but not sure what linter is good for Jekyll-based projects. @Rohith-Raju do you wanna take a look around it?

Though, hook or no-hook, contributors should disable their pre-existing linters before committing.

[the remark this answers is not in the document]
Comment (Rohith-Raju) [reply _sass/blog.scss:3]: @Nikhil-Ladha Sure, for now, I'll disable my linter and retry, Then I'll have a look around linters for Jekyll projects

**Comment by leecalcote** [comment]
@Rohith-Raju will you present these changes in today's meeting?

**Comment by asubedy** [comment]
@Rohith-Raju, the pattern image here has become really small, please have a look into this
image

File: _includes/card-modal.html [hunk 4]
```diff
@@ -3,30 +3,41 @@
...
     var btnClose = document.querySelectorAll('.close');
     var cont = document.querySelectorAll('.pattern')
+    var modal_container = document.querySelectorAll('.click-container')
 
+    for (i = 0; i < cont.length; i++) {
 
-    console.log(cont);
-    for(i=0;i<cont.length;i++){
-        console.log(cont[i]);
-        if(i%2!=0){
```
Comment (leecalcote) [inline _includes/card-modal.html:11]: Looking for modulus?

**Comment by Rohith-Raju** [comment]
@leecalcote learnt the hard way, should've removed the linter in the first place 😅 .// @asubedy

File: _sass/catalog.scss [hunk 5]
```diff
@@ -381,17 +411,108 @@
...
+    .modal-smp {
     width: 20px;
     bottom: 0;
     right: 0;
     vertical-align: middle;
     padding-right: 2px;
+}
+.modal-pattern-image {
+  width: 55%;
+  -o-object-fit: contain;
```
Comment (asubedy) [inline _sass/catalog.scss:425]: @Rohith-Raju what does this do?

File: _sass/blog.scss [hunk 6]
```diff
@@ -45,7 +45,6 @@
 		overflow: hidden;
 		a {
 			display: block;
-			width: 50%;
```
Comment (asubedy) [inline _sass/blog.scss:48]: Why was this removed?

File: _sass/catalog.scss (hunk 5 above)
Comment (Rohith-Raju) [reply _sass/catalog.scss:425]: @asubedy  "-o-object-fit: contain  = Specifies how the contents of a replaced element should be scaled relative to the box established by its used height and width.". Not entirely sure how it works. I didn't add this bust have picked up by the linter. I removed and looked for any differences, Saw none. So should I keep it or remove it

File: _sass/blog.scss (hunk 6 above)
Comment (Rohith-Raju) [reply _sass/blog.scss:48]: @asubedy navigation has to span the entire width of the container
With width:50%
image
When removed
image

Comment (asubedy) [reply _sass/blog.scss:48]: Oh ok nice nice. 👍

File: _sass/catalog.scss (hunk 5 above)
Comment (asubedy) [reply _sass/catalog.scss:425]: Yes @Rohith-Raju , please look carefully for any differences if there are none remove this.

**Review by asubedy — DISMISSED** [review]
LGTM 🚀

**Review by warunicorn19 — CHANGES_REQUESTED** [review]
The click outside listener doesn't seem to be working in safari, also could you check the padding inside the modal, I think it got increased because of some of the other CSS changes. Other than that everything else looks good.

**Comment by Rohith-Raju** [comment]
@warunicorn19 I don't have access to safari to experiment and resolve the issue, so I think @manikmmalhotra took care of that 
in this [pr](https://github.com/meshery/meshery.io/issues/772). And the padding inside the modal, Yes I'll look into that

**Comment by Rohith-Raju** [comment]
@warunicorn19 I tried looking for overriding styles but there are no other styles other than the styles from _catalog.scss being applied.

File: _sass/catalog.scss [hunk 7]
```diff
@@ -307,25 +300,62 @@
...
+    overflow-y: auto;
     width: 60%;
     top: 55%;
     left: 50%;
     transform: translate(-50%, -50%);
-    padding: 1.5em 2.5em 1.5em 3em;
+    border-radius: 15px;
     background: #fff;
-    border-radius: 1rem;
+    padding: 2rem 2rem 0 2rem;
```
Comment (warunicorn19) [inline _sass/catalog.scss:326]:
```suggestion
    padding: 1.5em 2.5em 1.5em 3em;
```

File: _sass/catalog.scss [hunk 8]
```diff
@@ -483,63 +604,188 @@
...
+  }
+  .modal-pattern-id {
+    position: absolute;
+    bottom: 0;
+    font-size: 8px;
+    right: 0;
+    padding-top: 10px;
+    margin-right: 14px;
+  }
+  .modal-window div div:not(:last-of-type) {
```
Comment (warunicorn19) [inline _sass/catalog.scss:664]:
```suggestion
```

File: _sass/catalog.scss [hunk 9]
```diff
@@ -483,63 +604,188 @@
...
+    margin-bottom: 15px;
```
Comment (warunicorn19) [inline _sass/catalog.scss:665]:
```suggestion
```

File: _sass/catalog.scss [hunk 10]
```diff
@@ -483,63 +604,188 @@
...
+  }
```
Comment (warunicorn19) [inline _sass/catalog.scss:666]:
```suggestion
```

**Review by warunicorn19 — APPROVED** [review]
Looking good! Thanks for seeing this through @Rohith-Raju
